### ISSUE TITLE
Feature Request: spatial local response normalization.

### ISSUE THREAD

**Issue by cesarsalgado** [tensorflow/tensorflow#1246]
Currently tf.nn.local_response_normalization only normalizes along the the depth. I would like to suggest adding a spatial version of LRN as in caffe: http://caffe.berkeleyvision.org/tutorial/layers.html#local-response-normalization-lrn

**Comment by cesarsalgado** [comment]
If someone is interested I made this implementation using python:

```python
def spatial_lrn(tensor, depth_radius=5, bias=1.0, alpha=1.0, beta=0.5):
  squared = tf.square(tensor)
  in_channels = tensor.get_shape().as_list()[3]
  kernel = tf.constant(1.0, shape=[depth_radius, depth_radius, in_channels, 1])
  squared_sum = tf.nn.depthwise_conv2d(squared,
                                       kernel,
                                       [1, 1, 1, 1],
                                       padding='SAME')
  bias = tf.constant(bias, dtype=tf.float32)
  alpha = tf.constant(alpha, dtype=tf.float32)
  beta = tf.constant(beta, dtype=tf.float32)
  return tensor / ((bias + alpha * squared_sum) ** beta)
```

**Comment by yash14123** [comment]
I would like to take this up @cesarsalgado

**Comment by cesarsalgado** [comment]
@yash14123 ok, thanks!

**Comment by siddharth-agrawal** [comment]
Is anyone working on this? I have not implemented an Op yet, so I think this could be a good learning experience for me. I can take this up, unless it needs to be done quickly. Can one of the admins comment?

**Comment by siddharth-agrawal** [comment]
@vrv Could you comment on this? (since you are already a participant on the issue)

**Comment by vrv** [comment]
@theaverageguy said he was working on it, but no movement in a while, so go ahead.  As always, good documentation and tests are going to be needed!

**Comment by siddharth-agrawal** [comment]
@vrv I tried doing this, I feel that I'm still not ready for writing an Op (I'm not familiar with Eigen functionalities, and don't understand what is efficient v/s what is not). I will look at some C++ issues before coming back to this. Are there some issues that will help me with this, or do you recommend another way?

**Comment by girving** [comment]
This is a reasonable feature, but I'll close for now since it's not getting any activity.  Happy to reopen if someone wants to take this on.
